contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,490 @@
+    
+        // If one side paid its fees, the ruling is in its favor. Note that if the other side had also paid, an appeal would have been created.
+        Round storage round = dispute.rounds[dispute.rounds.length - 1];
+        if (round.fundedSides.length == 1) finalRuling = round.fundedSides[0];
+
+        dispute.ruling = Decision(finalRuling);
+        dispute.resolved = true;
+
+        bytes32 claimId = dispute.claimId;
+        if (finalRuling == uint256(Decision.DismissResolution)) {            
+            hatArbitrator.dismissResolution(vault, claimId); //
+        } else {
+            // Arbitrator sided with committee or refused to arbitrate (gave 0 ruling).
+            hatArbitrator.executeResolution(vault, claimId);
+        }
+      
+        emit Ruling(IArbitrator(msg.sender), _disputeId, finalRuling);
+    }
+
+    /** @dev Submit a reference to evidence. EVENT.
+     *  @param _localDisputeId The id of the related dispute.
+     *  @param _evidenceURI Link to evidence.
+     */
+    function submitEvidence(uint256 _localDisputeId, string calldata _evidenceURI) external override {
+        DisputeStruct storage dispute = disputes[_localDisputeId];
+        // Note that by reading dispute's value we also check that it exists.
+        require(!dispute.resolved, "Dispute already resolved");
+
+        emit Evidence(klerosArbitrator, _localDisputeId, msg.sender, _evidenceURI);
+    }
+
+    // ************************ //
+    // *       Appeals        * //
+    // ************************ //
+
+    /** @dev Takes up to the total amount required to fund a side. Reimburses the rest. Creates an appeal if both sides are fully funded.
+     *  @param _localDisputeId The ID of the local dispute.
+     *  @param _side The option to fund. 0 - refuse to rule, 1 - make no changes, 2 - side with challenger.
+     *  @return fullyFunded Whether the side was fully funded or not.
+     */
+    function fundAppeal(uint256 _localDisputeId, uint256 _side) external payable override returns (bool) {
+        DisputeStruct storage dispute = disputes[_localDisputeId];
+        require(!dispute.resolved, "Dispute already resolved.");
+        require(_side <= RULING_OPTIONS, "Side out of bounds");
+
+        uint256 externalDisputeId = dispute.externalDisputeId;
+        (uint256 appealPeriodStart, uint256 appealPeriodEnd) = klerosArbitrator.appealPeriod(externalDisputeId);
+        require(block.timestamp >= appealPeriodStart && block.timestamp < appealPeriodEnd, "Appeal period is over.");
+
+        uint256 multiplier;
+        {
+            uint256 winner = klerosArbitrator.currentRuling(externalDisputeId);
+            if (winner == _side) {
+                multiplier = winnerMultiplier;
+            } else {
+                require(
+                    block.timestamp - appealPeriodStart <
+                        (appealPeriodEnd - appealPeriodStart).mulCap(loserAppealPeriodMultiplier) / MULTIPLIER_DIVISOR,
```

Comment:
as I understand it, you are comparing here an absolute timestamp with an interval, and so I would guess this checkw ould almost always fail. Or do I misunderstand?

Reply:
It's not an absolute timestamp but the difference between the current time and the time when the dispute started. We check whether we passed half of the appeal timeout or not